lang/cs/Org.Apache.REEF.Network/Elastic/Topology/Physical/Default/DefaultBroadcastTopology.cs

```diff
@@ -0,0 +1,128 @@
+// under the License.
+
+using Org.Apache.REEF.Network.Elastic.Config;
+using Org.Apache.REEF.Network.Elastic.Task.Impl;
+using Org.Apache.REEF.Tang.Annotations;
+using System.Collections.Generic;
+using Org.Apache.REEF.Common.Tasks;
+using System.Threading;
+using Org.Apache.REEF.Network.Elastic.Comm.Impl;
+using Org.Apache.REEF.Utilities.Logging;
+using System.Linq;
+using Org.Apache.REEF.Utilities.Attributes;
+using Org.Apache.REEF.Network.Elastic.Task;
+using Org.Apache.REEF.Network.Elastic.Failures;
+
+namespace Org.Apache.REEF.Network.Elastic.Topology.Physical.Default
+{
+    /// <summary>
+    /// Topology class managing data communication for broadcast operators.
+    /// </summary>
+    [Unstable("0.16", "API may change")]
+    internal sealed class DefaultBroadcastTopology : OneToNTopology
+    {
+        [Inject]
+        private DefaultBroadcastTopology(
+            [Parameter(typeof(OperatorParameters.StageName))] string stageName,
+            [Parameter(typeof(OperatorParameters.TopologyRootTaskId))] int rootId,
+            [Parameter(typeof(OperatorParameters.TopologyChildTaskIds))] ISet<int> children,
+            [Parameter(typeof(OperatorParameters.PiggybackTopologyUpdates))] bool piggyback,
+            [Parameter(typeof(OperatorParameters.OperatorId))] int operatorId,
+            [Parameter(typeof(TaskConfigurationOptions.Identifier))] string taskId,
+            [Parameter(typeof(GroupCommunicationConfigurationOptions.Retry))] int retry,
+            [Parameter(typeof(GroupCommunicationConfigurationOptions.Timeout))] int timeout,
+            [Parameter(typeof(GroupCommunicationConfigurationOptions.DisposeTimeout))] int disposeTimeout,
+            DefaultCommunicationLayer commLayer) : base(
+                stageName,
+                taskId,
+                Utils.BuildTaskId(stageName, rootId),
+                operatorId,
+                children,
+                piggyback,
+                retry,
+                timeout,
+                disposeTimeout,
+                commLayer)
+        {
+        }
+
+        public override DataMessage GetDataMessage<T>(int iteration, T[] data)
+        {
+            if (_piggybackTopologyUpdates)
+            {
+                return new DataMessageWithTopology<T>(StageName, OperatorId, iteration, data[0]);
+            }
+            else
+            {
+                return new DataMessage<T>(StageName, OperatorId, iteration, data[0]);
+            }
+        }
+
+        /// <summary>
+        /// Send a previously queued data message.
+        /// </summary>
+        /// <param name="cancellationSource">The source in case the task is cancelled</param>
+        protected override void Send(CancellationTokenSource cancellationSource)
+        {
+            ElasticGroupCommunicationMessage message;
+            int retry = 0;
+
+            // Check if we have a message to send
+            if (_sendQueue.TryPeek(out message))
```

Comment:
Yea. I think that I used this format because I thought was the convention in REEF. I think there will be  other part in the code that will show similar pattern.

Reply:
it's there just because some older REEF code was written before that syntax was available in C# :)

Reply:
Ah-ah gotcha. Fixed.